cmd/statechange/list.go

```diff
@@ -0,0 +1,192 @@
+package statechange
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/data-preservation-programs/singularity/cmd/cliutil"
+	"github.com/data-preservation-programs/singularity/database"
+	"github.com/data-preservation-programs/singularity/handler/statechange"
+	"github.com/data-preservation-programs/singularity/model"
+	"github.com/urfave/cli/v2"
+)
+
+var ListCmd = &cli.Command{
+	Name:    "list",
+	Aliases: []string{"ls"},
+	Usage:   "List deal state changes with optional filtering and pagination",
+	Description: `List deal state changes with comprehensive filtering options:
+- Filter by deal ID, state, provider, client address, and time range
+- Support for pagination and sorting
+- Export results to CSV or JSON formats`,
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "deal-id",
+			Usage: "Filter by specific deal ID",
+		},
+		&cli.StringFlag{
+			Name:  "state",
+			Usage: "Filter by deal state (proposed, published, active, expired, proposal_expired, rejected, slashed, error)",
+		},
+		&cli.StringFlag{
+			Name:  "provider",
+			Usage: "Filter by storage provider ID (e.g., f01234)",
+		},
+		&cli.StringFlag{
+			Name:  "client",
+			Usage: "Filter by client wallet address",
+		},
+		&cli.StringFlag{
+			Name:  "start-time",
+			Usage: "Filter changes after this time (RFC3339 format, e.g., 2023-01-01T00:00:00Z)",
+		},
+		&cli.StringFlag{
+			Name:  "end-time",
+			Usage: "Filter changes before this time (RFC3339 format, e.g., 2023-12-31T23:59:59Z)",
+		},
+		&cli.IntFlag{
+			Name:        "offset",
+			Usage:       "Number of records to skip for pagination",
+			DefaultText: "0",
+		},
+		&cli.IntFlag{
+			Name:        "limit",
+			Usage:       "Maximum number of records to return",
+			DefaultText: "100",
+		},
+		&cli.StringFlag{
+			Name:        "order-by",
+			Usage:       "Field to sort by (timestamp, dealId, newState, providerId, clientAddress)",
+			DefaultText: "timestamp",
+		},
+		&cli.StringFlag{
+			Name:        "order",
+			Usage:       "Sort order (asc, desc)",
+			DefaultText: "desc",
+		},
+		&cli.StringFlag{
+			Name:  "export",
+			Usage: "Export format (csv, json). If specified, results will be exported to a file instead of displayed",
+		},
+		&cli.StringFlag{
+			Name:  "output",
+			Usage: "Output file path for export (optional, defaults to statechanges-<timestamp>.csv/json)",
+		},
+	},
+	Action: func(c *cli.Context) error {
+		db, closer, err := database.OpenFromCLI(c)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		defer func() { _ = closer.Close() }()
```

Comment:
Why wrap the closer.Close function if you aren't going to do anything with the return value?